testserver/testserver.go

```diff
@@ -760,28 +760,33 @@ func (ts *testServerImpl) WaitForInit() error {
 }
 
 func (ts *testServerImpl) pollListeningURLFile(nodeNum int) error {
+	log.Printf("node %d: waiting for listening URL file %q\n", nodeNum, ts.nodes[nodeNum].listeningURLFile)
 	var data []byte
+	var err error
 	for i := 0; i < ts.serverArgs.pollListenURLTimeoutSeconds*10; i++ {
 		ts.mu.RLock()
 		state := ts.nodes[nodeNum].state
 		ts.mu.RUnlock()
 		if state != stateRunning {
-			return fmt.Errorf("server stopped or crashed before listening URL file was available")
+			return fmt.Errorf("node %d stopped or crashed before listening URL file was available", nodeNum)
 		}
-		var err error
 		data, err = os.ReadFile(ts.nodes[nodeNum].listeningURLFile)
-		if len(data) == 0 {
-			time.Sleep(100 * time.Millisecond)
-			continue
-		} else if err == nil {
+		if err == nil && len(data) > 0 {
 			break
-		} else if !os.IsNotExist(err) {
-			return fmt.Errorf("unexpected error while reading listening URL file: %w", err)
 		}
+		if err != nil && !os.IsNotExist(err) {
```

Comment:
Nit: left conjunct is a tautology, it can be dropped.

Reply:
It's not, when data is empty. I'll refactor a bit so it's less subtle.

TFTR!

Reply:
Right, 🤦 .